Update guidelines

seeforschauer opened the pull request.

AlekSi commented on CONTRIBUTING.md:
```diff
@@ -21,6 +22,44 @@ git remote add upstream https://github.com/FerretDB/FerretDB.git
 ### Setting up the development environment
 
+In the github-actions there are checks that new code complies the imports sorted convention
+and that the code is gofmt-ed. To avoid failing PR checks, please setup your beloved text editor
```
https://sembr.org

seeforschauer replied:
fixed, added to the guideline.

AlekSi replied:
I’m pretty sure we still break rule 4 of it. Please use one sentence per line in all markdown files except the changelog. See also #396

seeforschauer replied:
than you. Fixed sembr also for the `processes.md` file. Added pre-hook commit example.

AlekSi commented on CONTRIBUTING.md:
```diff
@@ -21,6 +22,44 @@ git remote add upstream https://github.com/FerretDB/FerretDB.git
 
 ### Setting up the development environment
 
+In the github-actions there are checks that new code complies the imports sorted convention
```
"github-actions" is a repository name. You probably mean "CI" there.

seeforschauer replied:
fixed

AlekSi commented on CONTRIBUTING.md:
```diff
@@ -21,6 +22,44 @@ git remote add upstream https://github.com/FerretDB/FerretDB.git
 
+In the github-actions there are checks that new code complies the imports sorted convention
+and that the code is gofmt-ed. To avoid failing PR checks, please setup your beloved text editor
+to gofmt files before push. Here provided some setup guides for VScode and Goland.
```
Might it be easier to just recommend `task fmt lint`?

seeforschauer replied:
I suggest giving several solutions that fits different people. 
With manual runs the risk of human factor "forgot to run" is higher.

seeforschauer replied:
Also added `task fmt lint` to the guideline)

AlekSi reviewed (commented):
`task docs-fmt`

AlekSi commented on CONTRIBUTING.md:
```diff
@@ -21,6 +21,48 @@ git remote add upstream https://github.com/FerretDB/FerretDB.git
+to gofmt files before push. Here provided some setup guides for VScode and Goland.
+You can also execute each time `task fmt lint` or project -> fmt functions of the text editor.
+
+After you edited *.md files, please run `task docs-fmt` to comply with <https://sembr.org>.
```
Unfortunately, we don’t have a linter for that yet; see #396

w84thesun commented on CONTRIBUTING.md:
```diff
@@ -21,6 +21,54 @@ git remote add upstream https://github.com/FerretDB/FerretDB.git
 
 ### Setting up the development environment
 
+In the CI there are checks that new code complies the imports sorted convention and that the code is gofmt-ed.
```
```suggestion
In the CI there are checks that new code complies with the imports sorting convention and that the code is gofmt-ed.
```

seeforschauer commented:
fixed, thank you.

AlekSi reviewed (commented):
As discussed, let's document `task fmt` and `task lint` targets, move most of the rest to the future blog post, and replace instructions for editors by links

seeforschauer commented:
> As discussed, let's document `task fmt` and `task lint` targets, move most of the rest to the future blog post, and replace instructions for editors by links

added doc, extracted to the blog post draft, at the moment no link, suppose to add it later after blog post will be completed.

seeforschauer commented:
@AlekSi @w84thesun PTAL

AlekSi commented on CONTRIBUTING.md:
```diff
@@ -19,25 +19,33 @@ cd FerretDB
+With `task` installed, you may run:
+
+* `task init` to install development tools.
+
```
[Markdown list](https://daringfireball.net/projects/markdown/syntax#list) should not have empty lines between items; it is both feels wrong for the reader of the raw text, and also renders worse on GitHub:

![image](https://user-images.githubusercontent.com/11512/165056494-fc6f2d84-90d4-4579-beec-edd44a78e0a0.png)

seeforschauer replied:
I checked it on Markdown linter and previewer locally and didn't find that wrong since it was more comfortable for me to read text when lines separated it.

AlekSi commented on CONTRIBUTING.md:
```diff
@@ -19,25 +19,33 @@ cd FerretDB
+
+* `task mongosh` to run `mongosh` to run commands against FerretDB.
+
+* `task fmt` to format source code.
```
What is missing there, as we discussed on the call, is the mention that there is the right time to change the source code. Steps before are about preparing for it, and steps after are about preparing for making a pull request.

seeforschauer replied:
Added step "change the code" - it was apparent that it happened somewhere here.

AlekSi reviewed (commented):
The PR title is outdated now

AlekSi commented on CONTRIBUTING.md:
````diff
@@ -19,25 +19,33 @@ cd FerretDB
 ```
 
-### Setting up the development environment
+### Run
````
(unless you mean https://youtu.be/FKYOWRJcJ2w?t=130)

seeforschauer replied:
changed to previous.

seeforschauer commented on .github/PROCESS.md:
```diff
@@ -16,3 +16,12 @@ We will do all that for you so you can focus on what is really interesting for y
    Autogenerated GitHub's squash commit's title and body should be manually replaced by PR's title and description; that includes both manual merges and auto-merges.
    Do not leave original commit messages in the squashed commit body.
+
+## Calls `to show and/or to explain`
```
changed

seeforschauer commented on .github/PROCESS.md:
```diff
@@ -16,3 +16,12 @@ We will do all that for you so you can focus on what is really interesting for y
+
+## Calls `to show and/or to explain`
+
+Keep guides short: we don't create rules; we automate.
```
moved upper, to common section.

seeforschauer commented on .github/PROCESS.md:
```diff
@@ -16,3 +16,12 @@ We will do all that for you so you can focus on what is really interesting for y
+
+Keep guides short: we don't create rules; we automate.
+
+Please multiply the knowledge shared on calls and 1-1 messages by writing a detailed article.
```
Added details that here a blog post meant.

seeforschauer commented on .github/PROCESS.md:
```diff
@@ -16,3 +16,12 @@ We will do all that for you so you can focus on what is really interesting for y
+Please multiply the knowledge shared on calls and 1-1 messages by writing a detailed article.
+1-1s are good to check what you might have missed along the way - what seemed to you obvious and to others is not.
+
+We encourage diversity between us as different points of view expand our vision and contribute to the project's success.
```
Moved to the process section

seeforschauer commented on CONTRIBUTING.md:
```diff
@@ -19,25 +19,33 @@ cd FerretDB
+### Run
 
-To run development commands, you should first install the [`task`](https://taskfile.dev/) tool.
+You should first install the [`task`](https://taskfile.dev/) tool to run development commands.
```
That was the change recommended by grammarly. Changed to previous.

seeforschauer commented on CONTRIBUTING.md:
```diff
@@ -19,25 +19,33 @@ cd FerretDB
-   This allows you to run commands against FerretDB.
+With `task` installed, you may run:
+
+* `task init` to install development tools.
```
Changed.  
Thank you for a reminder that it might be helpful to have a sequenced guideline for new people. 
For me, that were options to use, a set of instruments.
